records/2025-02-19.authmethods.md

```diff
@@ -0,0 +1,174 @@
+| SMS 2FA         | • Near-instant delivery<br>• Works on any mobile phone<br>• Ease of use<br>• Functions without internet<br>                                                                                                                                   | • Vulnerable to SIM swapping<br>• Costly<br>• International delivery issues<br>• SS7 network vulnerabilities<br>• No persistent storage<br>                                                 |
+| YubiKey         | • Highest security level (hardware-based)<br>• Immune to phishing attacks<br>• No batteries required<br>• Extremely durable<br>• Supports multiple protocols (FIDO2, U2F, OTP)<br>• Works offline<br>• No reliance on mobile network or email | • Initial cost ($45-60 per key)<br>• Requires physical possession<br>• Can be lost or damaged<br>• Limited mobile device support<br>• Learning curve for users<br>• May need backup key<br> |
+
+
+Unfortunately, we do not have the data to analyze how many departments provide YubiKeys (if at all) to their users.  
+Anecdotally, YubiKeys are not as intuitive to the average user compared to email or SMS.
+
+## Current auth flow
+```mermaid
+graph LR
+    A[User] --> B[Enter Password]
+    B --> C[2FA Authentication]
+
+    subgraph 2FA Options
+        C --> E[Email Code]
+        C --> F[SMS Code]
+        C --> G[YubiKey]
+    end
+
+    E --> D[Logged In User]
+    F --> D
+    G --> D
+```
+
+## Security Best Practices
+
+• Hardware security keys (like YubiKey) offer the strongest security   
+• Provide backup recovery methods   
+• Consider multiple authentication options for user flexibility   
+• If choosing between email and SMS, email generally offers better security against targeted attacks    
+
+# Recommendations
+
+The options below are not independent of each other. We can implement multiple options going forward.
+My recommendation is to give users options for different 2FA types. This way, users can choose what is convenient
+for them while we maintain control over the types of auth flows (for security purposes).
+
+
+## 1. Remove SMS as a 2FA option  
+
+Currently, Notify requires a user's phone number in the signup flow. 
+1. We can remove SMS as an option going forward.  
+2. We will need a migration plan to move existing SMS 2FA users to another auth_type.  
+3. We would then disable 2FA verification by SMS for all Notify users.  
+
+Benefits of removing SMS-based 2FA:  
+• Eliminates vulnerabilities to SIM swapping attacks  
+• SMS is known to be the least secure form of 2FA
+• Improves overall security posture
+
+
```

Comment:
We need disadvantages for removing SMS, with the most important con being the migration of users moving away from this authentication mechanism. Also, we need to remove phone numbers as required information part of the registration. We are losing one way to reach out to our users.